File: app/src/main/java/torille/fi/lurkforreddit/data/VideositeService.java
package torille.fi.lurkforreddit.data;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import torille.fi.lurkforreddit.data.models.jsonResponses.StreamableVideo;

/**
 * Created by eva on 9.4.2017.
 */


public interface VideositeService {

    interface Streamable {
        @GET("videos/{identifier}")
        Observable<StreamableVideo> getVideo(@Path("identifier") String identifier);
    }


}

File: app/src/main/java/torille/fi/lurkforreddit/data/VideositeService.kt
package torille.fi.lurkforreddit.data

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import torille.fi.lurkforreddit.data.models.jsonResponses.StreamableVideo

interface VideositeService {

    interface Streamable {
        @GET("videos/{identifier}")
        fun getVideo(@Path("identifier") identifier: String): Observable<StreamableVideo>
    }

}

